src/routes/scope-selection/ScopeSelectionScreen.tsx

```diff
@@ -1,44 +1,68 @@
+import normalize from 'normalize-strings'
+import React, { useLayoutEffect, useMemo, useState } from 'react'
+import { Pressable, ScrollView } from 'react-native'
 import styled, { useTheme } from 'styled-components/native'
 
 import Button from '../../components/Button'
 import DisciplineListItem from '../../components/DisciplineListItem'
 import Header from '../../components/Header'
+import HorizontalLine from '../../components/HorizontalLine'
 import RouteWrapper from '../../components/RouteWrapper'
+import SearchBar from '../../components/SearchBar'
 import ServerResponseHandler from '../../components/ServerResponseHandler'
-import { ContentSecondary } from '../../components/text/Content'
+import { ContentSecondary, ContentTextBold, ContentTextLight } from '../../components/text/Content'
 import { Heading } from '../../components/text/Heading'
 import { BUTTONS_THEME } from '../../constants/data'
 import { Discipline } from '../../constants/endpoints'
+import { formatDiscipline } from '../../hooks/helpers'
+import { useLoadAllDisciplines } from '../../hooks/useLoadAllDisciplines'
 import { useLoadDisciplines } from '../../hooks/useLoadDisciplines'
 import useReadSelectedProfessions from '../../hooks/useReadSelectedProfessions'
 import { RoutesParams } from '../../navigation/NavigationTypes'
-import { setSelectedProfessions } from '../../services/AsyncStorage'
-import { getLabels } from '../../services/helpers'
+import { pushSelectedProfession, setSelectedProfessions } from '../../services/AsyncStorage'
+import { getLabels, splitTextBySearchString } from '../../services/helpers'
 
-const DisciplineContainer = styled.View`
-  margin: 0 ${props => props.theme.spacings.sm};
+const HighlightContainer = styled.Text<{ disabled: boolean }>`
+  flex-direction: row;
+  color: ${props => (props.disabled ? props.theme.colors.disabled : props.theme.colors.black)};
 `
 
-const ButtonContainer = styled.View`
-  margin: ${props => props.theme.spacings.md} auto;
+const TextContainer = styled.View`
+  margin-top: ${props => props.theme.spacings.xxl};
+  margin-bottom: ${props => props.theme.spacings.lg};
 `
 
 const StyledText = styled(ContentSecondary)`
   text-align: center;
 `
 
-const TextContainer = styled.View`
-  margin-top: ${props => props.theme.spacings.xxl};
-  margin-bottom: ${props => props.theme.spacings.lg};
+const SearchContainer = styled.View`
+  margin: ${props => props.theme.spacings.sm};
 `
 
-const StyledScrollView = styled.ScrollView`
-  background-color: ${props => props.theme.colors.background};
+const ScopeContainer = styled.View`
+  margin: 0 ${props => props.theme.spacings.sm};
 `
 
+const ButtonContainer = styled.View`
+  margin: ${props => props.theme.spacings.md} auto;
+`
+
+export const searchProfessions = (disciplines: Discipline[] | undefined, searchKey: string): Discipline[] | undefined =>
+  disciplines?.filter(discipline =>
+    normalize(discipline.title).toLowerCase().includes(normalize(searchKey.toLowerCase()))
+  )
```

Comment:
I think this should go to a helper. Additionally, we should probably use our custom `normalizeSearchString` helper instead of plain `normalize`.